sites/friday/src/get-started/stacks/django.md

```diff
@@ -9,21 +9,113 @@ description: |
+required in order to have a successful deployment of Django on Upsun. Alternatively, you can read the blog post
+[_Up(sun) and running with Django_](https://upsun.com/blog/setting-up-django-on-upsun/) which walks through the steps
+required to run Django on Upsun.
+
+## `.environment`
+Django requires a few environment variables that have not been set. Open the `.environment` file and add the following:
+```shell
+export DJANGO_SETTINGS_MODULE=config.settings.production
+export DJANGO_SECRET_KEY="$PLATFORM_PROJECT_ENTROPY"
+export DJANGO_ALLOWED_HOSTS=".platformsh.site"
+```
+This section appends Django’s allowed hosts to include all URLs generated for Upsun preview environments, to update
+Django’s secret key to match the unique project hash, and to leverage production settings (in this case) across all
+Upsun environments.
+
+## `.upsun/config.yaml`
+Similarly, there are several changes we need to make to the Upsun configuration file so that Upsun knows how your Django
+application should behave. Open the `.upsun/config.yaml` file and make the changes below.
+### Start command
+Find the `web:commands:start` section. Change the following line from:
+```yaml
+start: "echo 'Put your web server command in here! You need to listen to \"$UNIX\" unix socket. Read more about it here: https://docs.upsun.com/create-apps/app-reference.html#web-commands'; sleep 60"
+```
+to
+```yaml
+start: "gunicorn -b unix:$SOCKET config.wsgi"
+```
+If your Django instance requires a different Web server, we also support [several other options](/languages/python/server.md).
+
+### Static assets
+In order to access Django's static assets we need to add a second location to the `web:locations` block. In the
+`.upsun/config.yaml` file, locate the `web:locations` section. Expand what is currently there to include a location for
+`/static`:
+
+```yaml
+            locations:
+              "/":
+                "passthru": true
+              "/static":
+                "allow": true
+                "expires": "1h"
+                "root": "static"
+```
 
-- [Upsun demo application](https://console.upsun.com/projects/create-project)
-- [Upsun Getting started guide](/get-started/here/_index.md)
-- [What is Upsun?](/learn/overview)
 
-## Documentation
+### Installation during the Build Hook
+Next we need to instruct Upsun to install our python and node dependencies. Scroll down a bit farther and find the
+`hooks:build` section. Change the following lines from:
+```yaml
+    build: |
+      set -eux
+
+      # Add build steps here.
+      # echo 'I am a build step'
+```
+to:
+```yaml
+    build: |
+      set -eux
+      pip install --upgrade pip
+      pip install -r requirements.txt
+      npm install
+      npm run build
+```
+### Deploy Hook
+Just below that, find the `deploy` section and change the following lines from:
+```yaml
+      deploy: |
+        set -eux
+        # echo 'Put your deploy command here'
+```
+to:
+```yaml
+            deploy: |
```

Comment:
indentation changed?

Reply:
maybe you need to display all top level keys above, like; 
```
applications:
    app:
        hooks:
            deploy: |
```

Reply:
and same for all other YAML snippets?

Reply:
that's the challenging part with yaml. do our code samples include the correct indentation according to where the keys we're changing are located so if someone copies and pastes they get the right indentation (assuming they're using 2 spaces per indent)? Or do we include the entire path (ie `applications:<app-name>:hooks`) and _assume_ the user knows that they cant copy and paste the whole block?

I don't have a good answer.